CookieConsentDialog component, usePrevious hook, gdprTracking configuration (#143)
newrelic/gatsby-theme-newrelic: packages/gatsby-theme-newrelic/README.md

};
```

> #### `gdprTracking`

Reviewer: A couple things here:

1) This configuration is not used in this PR, so having this listed in the README make confuse others. That being said, if we could only make the `gaTrackingId` configurable, then it would be great to list that here.
2) Can you make sure this gets added to the TOC above?

);
```

### `CookieConsentDialog`

A dialog box that pops up asking for cookie consent. This component renders at the bottom of the screen and provides options for accepting or denying cookies.

```js
import { CookieConsentDialog } from '@newrelic/gatsby-theme-newrelic';
```

**Example**

```js

> const myLayout = () => (

Reviewer: [nit] `Layout`

Figure it probably best to use React names to ensure we communicate that this likely goes in a layout _component_ (which are capitalized)

};
```

### `usePrevious`

> A hook that gets the previous state of a stateful value. Useful inorder to do comparing if the state has changed.

Reviewer: ```suggestion
A hook that gets the previous state of a stateful value. Useful to compare if the state has changed between render cycles.
```

```js
import { usePrevious } from '@newrelic/gatsby-theme-newrelic';
```

**Arguments**

> - `value` _(any)_: Value to track its' previous state.

Reviewer: ```suggestion
- `value` _(any)_: Value to track it's previous state.
```

Author: Actually I think it's neither of these, I think it is `its` 😆

Reviewer: Ha you're right! Apparently my brain was not working when I suggested this haha.

**Returns**

> `any` - The previous value to be computed.

Reviewer: ```suggestion
`any` - The previous value.
```

**Examples**

```js

> 

Reviewer: ```suggestion
```

const myComponent = () => {
  const [count, setCount] = useState(0);
  const prevCount = usePrevious(count);

  return (
    <div>
      <span>{`current count is ${count}, previous count is ${prevCount}`}

>       <button onClick ={()=>setCount(count+1)}>

Reviewer: ```suggestion
      <button onClick ={()=>setCount(count + 1)}>
```

    </div>
  )

> 

Reviewer: ```suggestion
```

}

> 

Reviewer: ```suggestion
```
